plugin/metrics/prometheus/factory.go

```diff
@@ -66,3 +67,29 @@ func (f *Factory) Initialize(logger *zap.Logger) error {
 func (f *Factory) CreateMetricsReader() (metricsstore.Reader, error) {
 	return prometheusstore.NewMetricsReader(f.options.Configuration, f.logger, f.tracer)
 }
+
+func (f *Factory) configureFromOptions(o *Options) {
```

Comment:
let's avoid this if there's no need to do anything other that assign a ptr, it can be done directly